osu.Server.Queues.ScoreStatisticsProcessor/Commands/Queue/ImportHighScoresCommand.cs

```diff
@@ -168,12 +169,26 @@ public async Task<int> OnExecuteAsync(CancellationToken cancellationToken)
                 }
             }
 
+            Console.WriteLine();
+            Console.WriteLine($"Sourcing from {highScoreTable} for {ruleset.ShortName} starting from {lastId}");
+            Console.WriteLine($"Inserting into scores and processing {(watchMode ? "indefinitely" : "as single run")}");
+
             if (watchMode)
             {
-                // Assume that `score_process_queue` is processed by a single process sequentially.
-                // We don't want to import until the score has finished processing or we may import incorrect pp data.
+                scoreStatisticsQueueProcessor = new ScoreStatisticsQueueProcessor();
+
+                if (SkipIndexing)
+                {
+                    throw new NotSupportedException("Skipping indexing in watch mode is not supported, "
```

Comment:
Hmm, this feels a bit weird. To confirm, we're *always* skipping indexing in watch mode because we delegate that to `osu-queue-score-statistics` right? Might be worth mentioning that either in an inline comment or in the exception message because right now it's not too clear (feels like it's saying "you can't skip indexing in this mode").

Reply:
I mean... yes the import job itself will not index, but the resulting score processing of the imported score _will_ index. So the net result of the operation will be that _an_ indexing _will_ happen. While the intent of the flag seemed that _no_ indexing should take place, not now, not later. So, also yes, the thing was supposed to be a "stop, this will do things you may not expect it to do".

If that is weird, or I misread intent with respect to the flag, I can remove this.